### PR TITLE
Rework Testsuite Engine

### PR DESCRIPTION
**Pull request by gjadi-mej**
- some rewording
- add code samples and links
- use tables instead of list to describe the various options

### CODE REVIEW

**Comment by frivieremicroej** [comment]
Link here

**Comment by frivieremicroej** [comment]
Please rebase to fix application & platform testsuite referenced links

File: ApplicationDeveloperGuide/testsuiteEngine.rst [hunk 1]
```diff
@@ -318,7 +427,7 @@ No dependency.
 Installation
 ------------
 
-This tool is a built-in platform tool.
```
Comment (frivieremicroej) [inline ApplicationDeveloperGuide/testsuiteEngine.rst:321]: Remove Dependencies & Installation legacy chapters and move 
"This tool is provided in the :ref:`Build Kit <mmm_build_kit>`." to introduction

Comment (gjadi-mej) [reply ApplicationDeveloperGuide/testsuiteEngine.rst:323]: Fixed.

**Pull request merged**